From 62b3641aa599fad4d5ee7ecbb7b83606b3a4b6fb Mon Sep 17 00:00:00 2001
From: Christian Dreher <cdreher@prophesee.ai>
Date: Fri, 15 Dec 2023 17:48:51 +0100
Subject: [PATCH 04/13] [MV-1764] Use the BufferAllocator in DataTransfer

This should not change any behavior, but it unfortunately change
the DataTransfer function signatures, breaking binary compatibility
with every plugin, making a software rebuild mandatory
---
 hal/cpp/include/metavision/hal/utils/data_transfer.h | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/hal/cpp/include/metavision/hal/utils/data_transfer.h b/hal/cpp/include/metavision/hal/utils/data_transfer.h
index e2edb04f1f..4e53981612 100644
--- a/hal/cpp/include/metavision/hal/utils/data_transfer.h
+++ b/hal/cpp/include/metavision/hal/utils/data_transfer.h
@@ -19,6 +19,7 @@
 #include <functional>
 
 #include "metavision/sdk/base/utils/object_pool.h"
+#include "metavision/hal/utils/buffer_allocator.h"
 
 namespace Metavision {
 
@@ -34,8 +35,11 @@ public:
     /// Alias for the type of the data transferred
     using Data = uint8_t;
 
+    /// Alias for the memory allocator for the data buffers
+    using Allocator = BufferAllocator<Data>;
+
     /// Alias for the type of the internal buffer of data
-    using Buffer = std::vector<Data>;
+    using Buffer = std::vector<Data, Allocator>;
 
     /// Alias for the object handling the buffers pool
     using BufferPool = SharedObjectPool<Buffer>;
-- 
2.39.2

